IMS.BL/Application/Program.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Linq;
+using System.Security.Policy;
+using IMS.BL.Database.DatabaseConnections.SqlDatabaseConnection;
+using IMS.BL.Domain.CustomExceptions;
+using IMS.BL.Domain.Services;
+using IMS.BL.Repositories;
+
+namespace IMS.BL.Application
+{
+    internal class Program
+    {
+        public static void Main(string[] args)
+        {
+            ProgramFlow();
+        }
+
+        private static void ProgramFlow()
+        {
+            Console.WriteLine("Please choose what operation you want to perform.");
+            Console.WriteLine("1 --> Add New Product.");
+            Console.WriteLine("2 --> Edit Existing Product.");
+            Console.WriteLine("3 --> Delete Existing Product.");
+            Console.WriteLine("4 --> Search For A Product.");
+            Console.WriteLine("5 --> Get All Products.");
+            Console.WriteLine("0 --> Exit.");
+            
+            int choice;
+            while (!int.TryParse(Console.ReadLine(), out choice))
+            {
+                Console.WriteLine("Please, enter a valid number from the choices above.");
+            }           
+            
+            // Required objects.
+            var sqlConnection = SqlConnectionProvider.Instance.SqlConnectionObject;
+            var sqlInventoryRepository = new SqlInventoryService(sqlConnection);
+            var inventoryRepository = new InventoryRepository();
+            inventoryRepository.SetInventoryRepository(sqlInventoryRepository);
+            var getProductData = new GetProductDataService();
+
+            try
+            {
+                switch (choice)
+                {
+                    case 1:
+                    {
+                        try
+                        {
+                            var productData = getProductData.GetProductToAdd();
+                            inventoryRepository.AddNewProduct(productData);
+
+                            Console.WriteLine($"The product has been added successfully!");
+                            break;
+                        }
+                        catch (Exception e)
+                        {
+                            throw new ProductException("Something went wrong while inserting new product! " +
+                                                       e.Message);
+                        }
```

Comment:
this is not a good way of catching exception, what if your exception is SQL provider connectivity issue and not ProductException, then why y are throwing a ProductException in that case ? as you did with the global handler you need to specify what exception do you expect to catch here, if there is any Exception other than the domain one you expect then let the global handler deals with them.